sdk/keyvault/azure-security-keyvault-administration/src/test/java/com/azure/security/keyvault/administration/KeyVaultCredentialPolicyTest.java

```diff
@@ -43,85 +52,120 @@ public class KeyVaultCredentialPolicyTest {
     private static final Flux<ByteBuffer> BODY_FLUX = Flux.defer(() ->
-        Flux.fromStream(
-            Stream.of(BODY.split(""))
-                .map(s -> ByteBuffer.wrap(s.getBytes(StandardCharsets.UTF_8)))
-        ));
-    private BasicAuthenticationCredential credential;
+        Flux.fromStream(Stream.of(BODY.split("")).map(s -> ByteBuffer.wrap(s.getBytes(StandardCharsets.UTF_8)))));
+
+    private HttpResponse unauthorizedHttpResponseWithWrongStatusCode;
     private HttpResponse unauthorizedHttpResponseWithHeader;
     private HttpResponse unauthorizedHttpResponseWithoutHeader;
+    private HttpResponse unauthorizedHttpResponseWithHeaderAndClaims;
     private HttpPipelineCallContext callContext;
     private HttpPipelineCallContext differentScopeContext;
     private HttpPipelineCallContext testContext;
     private HttpPipelineCallContext bodyContext;
     private HttpPipelineCallContext bodyFluxContext;
+    private BasicAuthenticationCredential credential;
+
+    private static HttpPipelineCallContext createCallContext(HttpRequest request, Context context) {
+        AtomicReference<HttpPipelineCallContext> callContextReference = new AtomicReference<>();
+
+        HttpPipeline callContextCreator = new HttpPipelineBuilder()
+            .policies((callContext, next) -> {
+                callContextReference.set(callContext);
+
+                return next.process();
+            })
+            .httpClient(ignored -> Mono.empty())
+            .build();
+
+        callContextCreator.sendSync(request, context);
+
+        return callContextReference.get();
+    }
 
     @BeforeEach
     public void setup() {
         HttpRequest request = new HttpRequest(HttpMethod.GET, "https://kvtest.vault.azure.net");
         HttpRequest requestWithDifferentScope = new HttpRequest(HttpMethod.GET, "https://mytest.azurecr.io");
 
-        HttpPipelineCallContext plainContext = createMockContext(request, null);
-
-        HttpPipelineCallContext differentScopeContext = createMockContext(requestWithDifferentScope, null);
+        Context bodyContextContext = new Context("KeyVaultCredentialPolicyStashedBody", BinaryData.fromString(BODY))
+            .addData("KeyVaultCredentialPolicyStashedContentLength", "21");
 
-        HttpPipelineCallContext testContext = createMockContext(request, null);
+        Context bodyFluxContextContext = new Context("KeyVaultCredentialPolicyStashedBody", BODY_FLUX)
+            .addData("KeyVaultCredentialPolicyStashedContentLength", "21");
 
-        HttpPipelineCallContext bodyContext = createMockContext(request, callContext -> {
-            callContext.setData("KeyVaultCredentialPolicyStashedBody", BinaryData.fromString(BODY));
-            callContext.setData("KeyVaultCredentialPolicyStashedContentLength", "21");
-        });
+        MockHttpResponse unauthorizedResponseWithWrongStatusCode = new MockHttpResponse(
+            new HttpRequest(HttpMethod.GET, "https://azure.com"), 500);
```

Comment:
I'm sorry to keep nagging about this but this name doesn't make sense. When the server responds 500, that isn't an "Unauthorized" response with the wrong status code; it's an "Internal Server Error" response with the right one.

Reply:
It's just an additional sanity check to make sure we don't let non-401s through. At this point I'll just delete the test altogether.